### ISSUE TITLE
Why the Nodegroup template use nested mapping instead of SSM recommended ami paramters?

### ISSUE THREAD

**Issue by tomiszili** [aws-quickstart/quickstart-amazon-eks#64]
Like this :
https://github.com/awslabs/amazon-eks-ami/blob/master/amazon-eks-nodegroup.yaml#L74-L77

**Comment by jaymccon** [comment]
This is partly due to legacy, from before this mapping was available, and also because we support ami's that are not available in ssm, currently only [Suse Linux](https://github.com/aws-quickstart/quickstart-amazon-eks/blob/master/templates/amazon-eks-nodegroup.template.yaml#L104), but we'd like to get more included.

Ideally we should use ssm where available and AMI mappings for the rest.

**Comment by tomiszili** [comment]
I'm trying to implement the SSM recommended parameter into CFN.
I modified the nodegroup template with:

```
 OStoK8sVersionMap:
    Amazon-Linux-HVM:
      EKS114: '{{resolve:ssm:/aws/service/eks/optimized-ami/1.14/amazon-linux-2/recommended/image_id}}'
    SUSE-SLES-15-HVM:
      EKS112: SLES15EKS112
      EKS112GPU: SLES15EKS112GPU
      EKS111: SLES15EKS111
      EKS111GPU: SLES15EKS111GPU
      EKS113: SLES15EKS113
      EKS113GPU: SLES15EKS113GPU
```

And commented out the AMZNEKS* keys.
The modified launch configuration's ImageId input looks like this:

 ```
ImageId: !If
        - UseCustomAmi
        - !Ref CustomAmiId
        - !If
          - AmznOS
          #- AWS::SSM::Parameter::Name</aws/service/eks/optimized-ami/1.14/amazon-linux-2/recommended/image_id>
          - !FindInMap [ OStoK8sVersionMap, !Ref NodeAMIOS, !FindInMap [ K8sVersionMap, !Ref KubernetesVersion, !FindInMap [ InstanceTypes, !Ref NodeInstanceType, Type ]]]
          - !FindInMap [ AWSAMIRegionMap, !Ref "AWS::Region", !FindInMap [ OStoK8sVersionMap, !Ref NodeAMIOS, !FindInMap [ K8sVersionMap, !Ref KubernetesVersion, !FindInMap [ InstanceTypes, !Ref NodeInstanceType, Type ]]]]
```
The CFN can't process this mapping inside the If function, the errors i get:
```Template error: Unable to get mapping for AWSAMIRegionMap::eu-central-1::{{resolve:ssm:/aws/service/eks/optimized-ami/1.14/amazon-linux-2/recommended/image_id}}```

If you have any idea how to solve this problem, please let me know. Thanks!

**Comment by tomiszili** [comment]
Any information about this enhancement?

**Comment by tomiszili** [comment]
This snippet could help to fetch the AMI ID from parameter names.
```
AMIFetchLambda:
    Type: AWS::Lambda::Function
    Properties:
      Handler: index.lambda_handler
      Role: !Ref AMIFetchLambdaRoleArn
      Runtime: python3.7
      Timeout: '30'
      Code:
        ZipFile: |
          import json
          import time
          import boto3
          import cfnresponse
          def lambda_handler(event, context):
              print('REQUEST BODY:n' + str(event))
              parameter_name = (event['ResourceProperties']['Parameter'][0])
              print(parameter_name)
              try:
                  responseData = {}
                  if event['RequestType'] == 'Delete':
                      print("delete")
                  else:
                      print(event['RequestType'])
                      client = boto3.client('ssm')
                      ami_id = client.get_parameter(Name=parameter_name)['Parameter']['Value']
                      print(ami_id)
                      responseData['AMI'] = ami_id
                  responseStatus = 'SUCCESS'
              except Exception as e:
                  print(e)
                  responseStatus = 'FAILED'
                  responseData = {'Failed': 'Something bad happened.'}
              cfnresponse.send(event, context, responseStatus, responseData)
```

**Comment by jaymccon** [comment]
wouldn't [dynamic references](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html) provide the same without the need for a custom resource ?

**Comment by tomiszili** [comment]
The public parameters like eks optimized amis are not supported.

> CloudFormation does not support using parameter labels or public parameters in dynamic references.

**Comment by tomiszili** [comment]
Now the canoncal Ubuntu images can be found with SSM. So the Lambda CFN snippet above can be used with these ssm parameter names.
```
    Amazon-Linux2-HVM:
      Code: /aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2
    Amazon-Linux-HVM:
      Code: /aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-gp2
    Ubuntu-Server-16.04-LTS-HVM:  
      Code: /aws/service/canonical/ubuntu/server/16.04/stable/current/amd64/hvm/ebs-gp2/ami-id
    Ubuntu-Server-18.04-LTS-HVM:  
      Code: /aws/service/canonical/ubuntu/server/18.04/stable/current/amd64/hvm/ebs-gp2/ami-id
```

**Comment by jaymccon** [comment]
implemented in #162
